JesusFilm/core
# refactor: admin theme

# Description

### Issue

Components needed for the resources hub need refactored out into the shared space to be used by both journeys-admin and watch. 

This is needed to access the journeys admin theme from outside the project for certain components.

[Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/37663178/todos/7452920587)

### Solution

This PR refactors the admin theme into the shared/ui space while leaving the shorthand `<ThemeProvider>` in the journeys-admin project to remain being used. Note the shared theme provider now handles the ThemeMode.journeysAdmin type.

# External Changes

Admin theme move meant that storybook needed moved to the shared/ui (which is needed for future component refactors).
Extend shared/ui/ThemeProvider to handle ThemeName.journeysAdmin
updated imports for linter

# Additional information

Big-bang discovery branch is here https://github.com/JesusFilm/core/pull/2810.

## How To Test
There should be no behavioural impact so ensure application still performs as expected.
One should be able to use the shared/ui Theme provider as shown inplace of the admin-journeys one.
<img width="325" alt="image" src="https://github.com/JesusFilm/core/assets/26043376/9537baab-c955-4d18-b115-813f189598aa">
The shared/ui ThemeProvider is used by journeys-admin, and watch so we would need to regression test these.
The journeys-admin ThemeProvider is used in journeys-admin so we might need to regression test this.
